vm/src/obj/objbyteinner.rs

```diff
@@ -0,0 +1,349 @@
+use crate::pyobject::PyObjectRef;
+
+use crate::function::OptionalArg;
+
+use crate::vm::VirtualMachine;
+
+use crate::pyobject::{PyResult, TypeProtocol};
+
+use crate::obj::objstr::PyString;
+use std::collections::hash_map::DefaultHasher;
+use std::hash::{Hash, Hasher};
+
+use super::objint;
+use super::objsequence::PySliceableSequence;
+use crate::obj::objint::PyInt;
+use num_traits::ToPrimitive;
+
+#[derive(Debug, Default, Clone)]
+pub struct PyByteInner {
+    pub elements: Vec<u8>,
+}
+
+impl PyByteInner {
+    pub fn new(
+        val_option: OptionalArg<PyObjectRef>,
+        enc_option: OptionalArg<PyObjectRef>,
+        vm: &VirtualMachine,
+    ) -> PyResult<PyByteInner> {
+        // First handle bytes(string, encoding[, errors])
+        if let OptionalArg::Present(enc) = enc_option {
+            if let OptionalArg::Present(eval) = val_option {
+                if let Ok(input) = eval.downcast::<PyString>() {
+                    if let Ok(encoding) = enc.clone().downcast::<PyString>() {
+                        if encoding.value.to_lowercase() == "utf8".to_string()
+                            || encoding.value.to_lowercase() == "utf-8".to_string()
+                        // TODO: different encoding
+                        {
+                            return Ok(PyByteInner {
```

Comment:
Could you run clippy on this? These seem like redundant `return`s.

Reply:
ok